app/jobs/reports/daily_registrations_report.rb

```diff
@@ -45,20 +48,32 @@ def report_body
       }
     end
 
+    # Queries for all the users that were ever created by date
+    # (includes users who were later deleted)
     def total_users
       params = {
         finish: finish,
-      }.transform_values { |v| ActiveRecord::Base.connection.quote(v) }
+      }.transform_values { |v| quote(v) }
 
       sql = format(<<-SQL, params)
         SELECT
           COUNT(*) AS total_users
-        , created_at::date AS date
-        FROM users
-        WHERE
-          created_at <= %{finish}
-        GROUP BY
-          created_at::date
+        , user_and_deleted_users.date AS date
+        FROM
+        (
+          SELECT created_at::date AS date
+          FROM users
+          WHERE users.created_at <= %{finish}
+
+          UNION ALL
```

Comment:
`UNION ALL` is probably not efficient, another approach would be to do the addition/joining in Ruby at the end, I could go either way. It's also a reporting query so I'm not overly concerned with efficiency? 🤷